index.sh

```diff
@@ -18,10 +18,80 @@
 set -e -u
 
 githubLatestTag() {
-  finalUrl=$(curl "https://github.com/$1/releases/latest" -s -L -I -o /dev/null -w '%{url_effective}')
-  printf "%s\n" "${finalUrl##*v}"
+  latestJSON="$( eval "$http 'https://api.github.com/repos/$1/releases/latest'" 2>/dev/null )" || true
+  if echo "$latestJSON" | grep 'API rate limit exceeded' >/dev/null 2>&1 ; then
+    # Don't leak and don't abuse please and thank you
+    ACCESS_TOKEN_DONT_LEAK="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"
```

Comment:
Can you say more about what this token is and where it came from?

If the issue is that running this script too many times from behind the same IP (I'm assuming the rate limit for unauthenticated requests is per-IP?), I think we can just set up a mirror; e.g. i'll set something up in the getmic.ro AWS account to automatically update an S3 object once a minute, and we can fall back to fetching the version number from that mirror.

Reply:
Alternatively, maybe we just implement a simple backoff-and-retry scheme? E.g. wait between 15 and 30 seconds (randomly selected) and then try again up to 3 times?

Reply:
This was a bit of code that got "left in" from when I was trying various solutions and trying to figure out the problem. The code works and it might help specific cases, so I saw no reason to remove it.

I've removed it now and committed my change.

This is an extra bit that provide compatibility in environments where the curl fallback can't be used because curl is unsupported AND getmicro hits the rate-limit due to being being mass-deployed or downloaded from crowded VPN. 

I created this public key which will never expire for the sole purpose and sole usage of the GitHub APIs. It can't be used to log into my account or anything else. If someone abuses it (which is doubtful as it would be far easier for them to generate more keys of their own), then we haven't lost much of anything.

Reply:
> Alternatively, maybe we just implement a simple backoff-and-retry scheme? E.g. wait between 15 and 30 seconds (randomly selected) and then try again up to 3 times?

This would be quite the complex scheme and something that would help hardly anyone (if anyone at all), so its doubtful that this would be worth the time and effort